haystack_experimental/components/generators/chat/openai.py

```diff
@@ -212,17 +200,14 @@ def from_dict(cls, data: Dict[str, Any]) -> "OpenAIChatGenerator":
         :returns:
             The deserialized component instance.
         """
-        deserialize_secrets_inplace(data["init_parameters"], keys=["api_key"])
-        init_params = data.get("init_parameters", {})
-        serialized_callback_handler = init_params.get("streaming_callback")
-        if serialized_callback_handler:
-            data["init_parameters"]["streaming_callback"] = deserialize_callable(serialized_callback_handler)
+        component = super(OpenAIChatGenerator, cls).from_dict(data)
```

Comment:
I don't think we can do this as `OpenAIChatGeneratorBase.from_dict` will return a component of type `haystack.components.generators.OpenAIChatGenerator`, breaking the class hierarchy. We'll have to keep the original version of the class method.

Reply:
Huh, that's interesting - I would've thought otherwise (I guess passing `cls` to `super` avoids the type issue I mentioned). 

In any event, it looks like it works primarily because the new params are optional. Let's keep the full method all the same.

Reply:
done